src/apps/the-exchange/components/EnterAmount/EnterAmount.tsx

```diff
@@ -29,72 +62,155 @@ const EnterAmount = ({ type, tokenSymbol }: EnterAmountProps) => {
             toTokenAddress: receiveToken?.address || '',
             toChainId: receiveToken?.chainId || 0,
             toTokenDecimals: receiveToken?.decimals || 0,
+            slippage: 0.05,
+        };
+
+        try {
+            const offer = await getBestOffer(params);
+
+            if (offer && receiveToken) {
+                const usdPrice = await getPrice(receiveToken.address, receiveToken.chainId);
+                if (usdPrice) {
+                    setAmountReceive({
+                        tokenAmount: offer.tokenAmountToReceive,
+                        usdAmount: usdPrice.usd * offer.tokenAmountToReceive,
+                    });
+                } else {
+                    setAmountReceive({ tokenAmount: offer.tokenAmountToReceive, usdAmount: 0 });
+                }
+            } else {
+                setIsNoOffer(true);
+            }
+
+            return offer;
+        } catch (error) {
+            console.error('Failed to fetch the best offer');
+            return null;
+        } finally {
+            setIsOfferLoading(false);
         }
-        const offer = await getBestOffer(params);
+    };
 
-        return offer;
-    }
-    
-    useEffect(() => {
-        getOffer().then((offer) => setBestOffer(offer));
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [amountSwap, swapToken, receiveToken])
+    // Debounced getOffer function with 2-second delay
+    const debouncedGetOffer = _.debounce(() => {
+        getOffer().then((offer) => {
+            if (offer) setBestOffer(offer);
+        });
+    }, 2000);
+
+    // Similar function being used for the SendModal and AccountModal to check tokens balances
+    const tokenBalanceLimit = (tokenAmount: number) => {
+        if (!swapToken) return;
+
+        const assetBalance =
+            swapToken.chainId &&
+            balances[swapToken.chainId]?.[walletAddress as string]?.find((balance) => {
+                if (!swapToken.address) {
+                    setErrorMessage('This token is not being supported by your wallet');
```

Comment:
Is this a better message maybe?

```suggestion
                    setErrorMessage('This token does not exist in your wallet');
```

Reply:
Yes, actually it is haha, thanks!  Changed